### ISSUE TITLE
Incomplete/corrupt Clojars download for rewrite-clj makes lein unrunnable

### ISSUE THREAD

**Issue by coyotesqrl** [weavejester/cljfmt#237]
I am still not sure this isn't just my issue, but I have poked at it a good deal today and last week with the same result.

I've added the lein plugin to my `project.clj` and when I try to run `lein cljfmt check` or `lein cljfmt fix`, I get a compilation error - `clojure.lang.Compiler$CompilerException: Syntax error compiling at (cljfmt/core.cljc:75:3).`

with an underlying cause of: `Caused by: java.lang.RuntimeException: No such var: z/insert-space-right`

Taking a look in my local maven repository in the `rewrite-clj/rewrite-clj/1.0.605-alpha` directory, I see that I have only three files, the remote repo meta file, the pom, and the sha for the pom (the sha does match). There is no jar.

I've tried manually deleting the directory, in case I just had a network issue during download, but when I run again I get the same contents.

I am pretty sure that this point that the artifact on Clojars is just incomplete.

Assuming they are api-compatible, I'd suggest upgrading to the newer 1.0.644-alpha version.

**Comment by weavejester** [comment]
I cleared my Maven cache, created a new project, added `[lein-cljfmt "0.8.0"]` as a plugin, and it works fine.

I don't think rewrite-clj is corrupt or incomplete. I think you're running into a dependency issue where an older version of rewrite-clj is in your dependency tree, and that version doesn't have `z/insert-space-right`.

**Comment by coyotesqrl** [comment]
I was afraid it might end up being something on my end that I'd been unable to find as of yet. I'll keep digging.

**Comment by coyotesqrl** [comment]
I have no idea where the dependency is getting pulled it, but based on your comment, I pulled out the plugin and added it with an alias to run under my user profile. Then I specifically added the right version of rewrite-clj as a dependency in that profile as well.

Good enough for now.

Thanks for your help.

**Comment by marksto** [comment]
@coyotesqrl, could you, please, share the profile you combined (or at least the "right version of rewrite-clj" dependency)?
I'm running into the same issue right now.

**Comment by marksto** [comment]
Ok, I guess it's `lein-kibit` that pulls that conflicting old version of the `rewrite-clj`.

**Comment by pedro-w** [comment]
I am also having this problem; @marksto @coyotesqrl  please could you explain the fix? I didn't know that the deps would be resolved in such a way that different plugins couldn't have different versions of their dependencies.
Thanks!

**Comment by Chouser** [comment]
I had an ancient version of `lein-ancient` (go figure) in my `/.lein/profiles.clj` that was causing me this problem.

I learned that by running `lein deps :plugin-tree`. So I commented out `lein-ancient` in my lein profiles.clj, ran that again, and could see the version desired by `[lein-cljfmt "0.8.0"]` is `[rewrite-clj "1.0.605-alpha"]`

For now, I'm just leaving lein-anceint commented-out and cljfmt is working fine.
